libtenzir/src/tql2/eval_unary.cpp

```diff
@@ -151,6 +151,20 @@ auto evaluator::eval(const ast::unary_expr& x) -> multi_series {
     X(neg);
     X(not_);
 #undef X
+    case move: {
+      if (ast::field_path::try_from(x.expr)) {
+        diagnostic::warning("move is not supported here")
+          .primary(x.op, "has no effect")
+          .hint("move only works on fields within assignments")
+          .emit(ctx_);
+      } else {
+        diagnostic::warning("move has no effect")
+          .primary(x.expr, "is not a field")
+          .hint("move only works on fields within assignments")
+          .emit(ctx_);
```

Comment:
FWIW, you could make this check at parse-time. When parsing `move`, you can check whether the next expression is a `field_path`, and otherwise error. You could also give `move` it's own `expression_kind` and let it wrap the `field_path` directly, such that you encode this property into the type system. Just as an idea, we don't necessarily have to do this.

Reply:
I'll write a follow-up story for this one.

Reply:
See https://github.com/tenzir/issues/issues/2964